packages/checkout/widgets-lib/src/widgets/bridge/components/WalletAndNetworkSelector.tsx

```diff
@@ -90,28 +86,15 @@ export function WalletAndNetworkSelector() {
     return getWalletProviderNameByProvider(toWalletWeb3Provider);
   }, [toWalletWeb3Provider]);
 
-  const fromWalletSelectorOptions = useMemo(() => {
-    const options = [WalletProviderName.METAMASK.toString()];
-    if (checkout.passport) {
-      options.push(WalletProviderName.PASSPORT.toString());
-    }
-    if (isWalletConnectEnabled) {
-      options.push('walletconnect');
-    }
-    return options;
-  }, [checkout, isWalletConnectEnabled]);
+  const fromWalletSelectorOptions = useMemo(() => providers, [providers]);
 
-  const toWalletSelectorOptions = useMemo(() => {
-    const options = [WalletProviderName.METAMASK.toString()];
-
-    if (checkout.passport && fromNetwork === l1NetworkChainId) {
-      options.push(WalletProviderName.PASSPORT.toString());
-    }
-    if (isWalletConnectEnabled) {
-      options.push('walletconnect');
-    }
-    return options;
-  }, [checkout, fromNetwork, fromWalletProviderName, isWalletConnectEnabled]);
+  const toWalletSelectorOptions = useMemo(() => (
+    providers
+      .filter((providerDetail) => (
+        providerDetail.info.rdns !== WalletProviderRdns.PASSPORT
+        || (providerDetail.info.rdns === WalletProviderRdns.PASSPORT && fromNetwork === l1NetworkChainId)
```

Comment:
Are we assuming here that we're not using passport injected into the checkout instance anymore?

Reply:
I've integrated `passport` into the useInjectedProviders hook, so it will use the injected passport provider via EIP6963 .. if it can't resolve that then it attempts a fallback to the injected passport instance provider. And then just simulates the EIP6963 data so it should be future proof.